core/src/main/scala/io/chrisdavenport/shellfish/syntax/path/package.scala

```diff
@@ -0,0 +1,374 @@
+     * Saves the content at the end of the file in form of a String.
+     *
+     * @param path
+     *   The path to write to
+     * @param contents
+     *   The contents to write to the file
+     * @return
+     *   The number of bytes written
+     */
+    def append(contents: String): IO[Long] = FilesOs.append(path)(contents)
+
+    /**
+     * Similar to `write`, but appends to the file instead of overwriting it.
+     * Saves the content at the end of the file in form of a ByteVector.
+     *
+     * @param path
+     *   The path to write to
+     * @param contents
+     *   The contents to write to the file
+     * @return
+     *   The number of bytes written
+     */
+    def appendBytes(contents: ByteVector): IO[Long] =
+      FilesOs.appendBytes(path)(contents)
+
+    /**
+     * Similar to `write`, but appends to the file instead of overwriting it.
+     * Saves each line of the content at the end of the file in form of a Vector
+     * of Strings.
+     *
+     * @param path
+     *   The path to write to
+     * @param contents
+     *   The contents to write to the file
+     * @return
+     *   The number of bytes written
+     */
+    def appendLines(contents: Vector[String]): IO[Long] =
+      FilesOs.appendLines(path)(contents)
+
+    /**
+     * Similar to `write`, but appends to the file instead of overwriting it.
+     *
+     * @param A
+     *   The type of the contents to write
+     * @param path
+     *   The path to write to
+     * @param contents
+     *   The contents to write to the file
+     * @param codec
+     *   The codec that translates the type A into a ByteVector
+     * @return
+     *   The number of bytes written
+     */
+    def appendAs[A](contents: A)(implicit codec: Codec[A]): IO[Long] =
+      FilesOs.appendAs[A](path)(contents)
+
+  }
+
+  private val files = Files[IO]
+
+  implicit class FileOps(val path: Path) extends AnyVal {
+
+    /**
+     * Copies the source to the target, failing if source does not exist or the
+     * target already exists. To replace the existing instead, use
+     * `source.copy(CopyFlags(CopyFlag.ReplaceExisting))(target)`.
+     */
+    def copy(to: Path): IO[Unit] = files.copy(path, to)
+
+    /**
+     * Copies the source to the target, following any directives supplied in the
+     * flags. By default, an error occurs if the target already exists, though
+     * this can be overriden via CopyFlag.ReplaceExisting.
+     */
+    def copy(flags: CopyFlags)(to: Path): IO[Unit] = files.copy(path, to, flags)
+
+    /**
+     * Creates the specified directory. Fails if the parent path does not
+     * already exist.
+     */
+    def createDirectory: IO[Unit] = files.createDirectories(path)
+
+    /**
+     * Creates the specified directory with the specified permissions. Fails if
+     * the parent path does not already exist.
+     */
+    def createDirectory(permissions: Option[Permissions]): IO[Unit] =
+      files.createDirectories(path, permissions)
```

Comment:
I don't remember how it works precisely but IF passing a `None` in permissions blanks out all the previous permissions on the file, I'll specify it in the scala doc.
If this is not the case I'll refactor the method as follows:
```suggestion
    def createDirectory(permissions: Permissions): IO[Unit] =
      files.createDirectories(path, permissions.some)
```
As `createDirectory(None)` will be functionally equivalent to `createDirectory`

Reply:
I wasn't able to get the permissions using the JFiles API in Windows, but in Linux (maybe as well as in macos) using `createFile` and `createDirectory` respectively output this: 

```
rw-rw-r--
rwxrwxr-x
```
Thus these are the defaults